src/containers/User/Login.tsx

```diff
@@ -1,66 +1,135 @@
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
 import { useRouter } from '../../utils/hooks/useRouter'
 import { useUserState } from '../../contexts/UserState'
 import { Link } from 'react-router-dom'
-import { Form, Button, Container, Grid, Segment, Header } from 'semantic-ui-react'
+import { Form, Button, Container, Grid, Segment, Header, Dropdown } from 'semantic-ui-react'
 import config from '../../config.json'
 import isLoggedIn from '../../utils/helpers/loginCheck'
 import strings from '../../utils/constants'
-import setUserInfo from '../../utils/helpers/fetchUserInfo'
+import { postRequest as attemptLogin } from '../../utils/helpers/fetchMethods'
+import { User, LoginPayload, OrganisationSimple, TemplatePermissions } from '../../utils/types'
+
+const loginURL = config.serverREST + '/login'
+const loginOrgURL = config.serverREST + '/login-org'
 
 const Login: React.FC = () => {
   const [username, setUsername] = useState('')
   const [password, setPassword] = useState('')
   const [isError, setIsError] = useState(false)
+  const [user, setUser] = useState<User>()
+  const [orgList, setOrgList] = useState<OrganisationSimple[]>([])
+  const [JWT, setJWT] = useState<string>()
+  const [templatePermissions, setTemplatePermissions] = useState<TemplatePermissions>()
+  const [selectedOrgIndex, setSelectedOrgIndex] = useState(0)
   const { push, history } = useRouter()
-  const { login } = useUserState()
+  const { onLogin } = useUserState()
+
+  // useEffect ensures isLoggedIn only runs on first mount, not re-renders
+  useEffect(() => {
+    if (isLoggedIn()) push('/')
+  }, [])
+
+  const handleSubmit = async () => {
+    if (!user) {
+      // User login
+      const loginResult: LoginPayload = await attemptLogin({ username, password }, loginURL)
+
+      if (!loginResult.success) setIsError(true)
+      else {
+        setIsError(false)
+        setUser(loginResult.user)
+        setJWT(loginResult.JWT)
+        setTemplatePermissions(loginResult.templatePermissions)
+        setOrgList(loginResult?.orgList as OrganisationSimple[])
+      }
```

Comment:
Consider moving this to a `attemptLogin` that has the `const loginURL = config.serverREST + '/login'` also. And uses the `fetchMethod` utility plus receive the `onSuccess` and `onFailure` methods to set up the same as what is done here in the if...else statement. Also, consider using a single state for setting (user, JWT, templatePermissions and orgList).